external/kafka-0-10-sql/src/main/scala/org/apache/spark/sql/kafka010/KafkaDataConsumer.scala

```diff
@@ -279,9 +279,8 @@ private[kafka010] case class InternalKafkaConsumer(
       if (record.offset > offset) {
         // This may happen when some records aged out but their offsets already got verified
         if (failOnDataLoss) {
-          reportDataLoss(true, s"Cannot fetch records in [$offset, ${record.offset})")
```

Comment:
It's just breaks the whole concept. When `failOnDataLoss ` enabled this exception means it should fail.

Reply:
@gaborgsomogyi Thank you Gaborgsomogyi for looking at it. For the batch queries, it will always fail if it fails to read any data from the provided offsets due to lost data.  https://spark.apache.org/docs/latest/structured-streaming-kafka-integration.html . With this change,it wont fail .Instead It will return all the available messages within the requested offset range.